src/com/superliminal/magiccube4d/PolytopePuzzleDescription.java

```diff
@@ -266,15 +268,50 @@ private static void Assumpt(boolean condition) {
      * omnitruncated regular
      */
 
-    public PolytopePuzzleDescription(String schlafliProduct,
-        double length, // usually int but can experiment with different cut depths
-        ProgressCallbacks progress)
+    public PolytopePuzzleDescription(String prescription, ProgressCallbacks progress)
     {
-        this.schlafliProduct = schlafliProduct;
-        this.edgeLength = length;
+        // Caller may have passed us "Grand Antiprism" from a menu,
+        // but the lower level function we're going go call doesn't allow spaces,
+        // and understands Grand_Antiprism as a special case, so convert.
+        prescription = prescription.replaceAll("Grand Antiprism",
```

Comment:
Is this the only special case? It might be nice to siphen this functionality off to a separate function.

Reply:
Are you suggesting a separate private function that this ctor would call, or a separate preprocessing function that the caller would call?

I guess this is the only special case, and I'm not proud of it.  It's basically automagic so the caller can be artificially simpler, but in reality adds quite a lot of entropy.  E.g. it makes it so that the basic filter for "does this string look like a legal prescription?" no longer is expressible by a straightforward regex.

Reply:
I had just imagined a separate private function, but definitely unnecessary if this will be the only case.

I am a little surprised there aren't more special-case names given all the polytopes you support. Maybe in the future more special names could be supported and this could be reconsidered then.